proto/external_storage.proto

```diff
@@ -0,0 +1,72 @@
+syntax = "proto3";
+// purposely leaving out delete. that is up to individual external
+// storage server operators to do on their end
+
+service ProxyStorage {
+
+    rpc GetModuleModfile (ModuleMetadata) returns (ModuleMod) {}
```

Comment:
I think `GetModuleModfile` is a little verbose, maybe `Modfile` or `GetModfile`? Same for the methods below.

Reply:
good point! I reduced redundancy in this and all the other functions